File: mirai-console/backend/mirai-console/src/com/llamalab/safs/FileSystem.java
/*
 * Copyright (C) 2019 Henrik Lindqvist
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.llamalab.safs;

import com.llamalab.safs.attribute.UserPrincipalLookupService;
import com.llamalab.safs.spi.FileSystemProvider;

import java.io.Closeable;
import java.io.IOException;
import java.util.Set;

public abstract class FileSystem implements Closeable {
  public abstract FileSystemProvider provider ();
  public abstract boolean isOpen ();
  public abstract boolean isReadOnly ();
  public abstract Set<String> supportedFileAttributeViews();
  public abstract String getSeparator ();
  public abstract Path getPath (String first, String... more);
  public abstract PathMatcher getPathMatcher (String syntaxAndPattern);
  public abstract Iterable<FileStore> getFileStores ();
  public abstract Iterable<Path> getRootDirectories ();
  public abstract UserPrincipalLookupService getUserPrincipalLookupService ();
  public abstract WatchService newWatchService () throws IOException;
}

File: mirai-core-api/src/commonMain/kotlin/utils/java.bindings/nio/FileSystem.kt
/*
 * Copyright 2019-2020 Mamoe Technologies and contributors.
 *
 *  此源代码的使用受 GNU AFFERO GENERAL PUBLIC LICENSE version 3 许可证的约束, 可以在以下链接找到该许可证.
 *  Use of this source code is governed by the GNU AGPLv3 license that can be found through the following link.
 *
 *  https://github.com/mamoe/mirai/blob/master/LICENSE
 */

@file:Suppress(
    "unused",
    "NO_ACTUAL_FOR_EXPECT",
    "PackageDirectoryMismatch",
    "NON_FINAL_MEMBER_IN_FINAL_CLASS",
    "VIRTUAL_MEMBER_HIDDEN",
    "RedundantModalityModifier",
    "REDUNDANT_MODIFIER_FOR_TARGET",
    "REDUNDANT_OPEN_IN_INTERFACE"
)

/**
 * Bindings for JDK.
 *
 * All the sources are copied from OpenJDK. Copyright OpenJDK authors.
 */

package java.nio.file

import java.io.Closeable

public expect abstract class FileSystem : Closeable {
    //public abstract fun provider(): java.nio.file.spi.FileSystemProvider
    public abstract fun close()
    public abstract fun isOpen(): Boolean
    public abstract fun isReadOnly(): Boolean
    public abstract fun getSeparator(): String
    public abstract fun getRootDirectories(): Iterable<Path>

    //public abstract fun getFileStores(): Iterable<FileStore>
    public abstract fun supportedFileAttributeViews(): Set<String>
    public abstract fun getPath(first: String, vararg more: String): Path
    //public abstract fun getPathMatcher(syntaxAndPattern: String): PathMatcher
    //public abstract fun getUserPrincipalLookupService(): UserPrincipalLookupService
    // public abstract fun newWatchService(): WatchService
}  